sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/kafka/AbstractKafkaPropertiesBeanPostProcessor.java

```diff
@@ -0,0 +1,306 @@
+            ApiVersionsRequest apiVersionsRequest = new ApiVersionsRequest.Builder().build();
+            ApiVersionsRequestData apiVersionsRequestData = (ApiVersionsRequestData) ReflectionUtils.invokeMethod(dataMethod, apiVersionsRequest);
+            if (apiVersionsRequestData != null) {
+                String clientSoftwareName = apiVersionsRequestData.clientSoftwareName();
+                if (clientSoftwareName != null && !clientSoftwareName.contains(AZURE_SPRING_EVENT_HUBS_KAFKA_OAUTH)) {
+                    apiVersionsRequestData.setClientSoftwareName(apiVersionsRequestData.clientSoftwareName()
+                        + AZURE_SPRING_EVENT_HUBS_KAFKA_OAUTH);
+                    apiVersionsRequestData.setClientSoftwareVersion(VERSION);
+                }
+            }
+        }
+    }
+
+    /**
+     * Detect whether we need to configure SASL/OAUTHBEARER properties for {@link KafkaProperties}. Will configure when
+     * the security protocol is not configured, or it's set as SASL_SSL with sasl mechanism as null or OAUTHBEAR.
+     *
+     * @param sourceProperties the source kafka properties for admin/consumer/producer to detect
+     * @return whether we need to configure with Spring Cloud Azure MSI support or not.
+     */
+    boolean needConfigureSaslOAuth(Map<String, Object> sourceProperties) {
+        return meetAzureBootstrapServerConditions(sourceProperties) && meetSaslOAuthConditions(sourceProperties);
+    }
+
+    private boolean meetSaslOAuthConditions(Map<String, Object> sourceProperties) {
+        String securityProtocol = (String) sourceProperties.get(SECURITY_PROTOCOL_CONFIG);
+        String saslMechanism = (String) sourceProperties.get(SASL_MECHANISM);
+        String jaasConfig = (String) sourceProperties.get(SASL_JAAS_CONFIG);
+        if (meetSaslProtocolConditions(securityProtocol) && meetSaslOAuth2MechanismConditions(saslMechanism)
+            && meetJaasConditions(jaasConfig)) {
+            return true;
+        }
+        getLogger().info(LOG_OAUTH_AUTOCONFIGURATION_RECOMMENDATION, saslMechanism);
+        return false;
+    }
+
+    private boolean meetSaslProtocolConditions(String securityProtocol) {
+        return securityProtocol == null || SECURITY_PROTOCOL_CONFIG_SASL.equalsIgnoreCase(securityProtocol);
+    }
+
+    private boolean meetSaslOAuth2MechanismConditions(String saslMechanism) {
+        return saslMechanism == null || SASL_MECHANISM_OAUTH.equalsIgnoreCase(saslMechanism);
+    }
+    private boolean meetJaasConditions(String jaasConfig) {
+        if (jaasConfig == null) {
+            return true;
+        }
+        AtomicBoolean flag = new AtomicBoolean(false);
+        JaasResolver resolver = new JaasResolver();
+        resolver.resolve(jaasConfig).ifPresent(jaas -> {
+            if (AZURE_CONFIGURED_JAAS_OPTIONS_VALUE.equals(jaas.getOptions().get(jaas.getOptions()))) {
+             flag.set(true);
+            }
+        });
+        return flag.get();
+    }
```

Comment:
can we use 

```java
return resolver.resolve(jaasConfig).map(AZURE_CONFIGURED_JAAS_OPTIONS_VALUE.equals(jaas.getOptions().get(jaas.getOptions()))).orElse(false);
```

Reply:
why this `jaas.getOptions().get(jaas.getOptions())`?

Reply:
I will update, `jaas.getOptions().get(jaas.getOptions())` is a typo and also the reason why the ut fails now.